cmake/zpp.cmake

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024 Basalte bv
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Generate compile_commands.json in the output directory
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE CACHE BOOL
+    "Export CMake compile commands. Used by zpp.py script"
+    FORCE
+)
```

Comment:
this is already set in `cmake/modules/kernel.cmake`

Reply:
I copied it from there, it should be merged somewhere indeed, but for the PoC it was easier to have it here with less change of getting conflicts.